esmvalcore/preprocessor/_bias.py

```diff
@@ -106,7 +106,7 @@ def bias(products, bias_type='absolute', denominator_mask_threshold=1e-3,
         cube.metadata = cube_metadata
         cube.units = new_units
         product.attributes['units'] = new_units
-        product.wasderivedfrom(reference_product)
+        product.wasderivedfrom(reference_product.copy_provenance())
```

Comment:
This copy is not needed, since `reference_product` is not updated in this loop: https://github.com/ESMValGroup/ESMValCore/blob/2ecbb5c3cf183b3298b840b9187a5039529f6b39/esmvalcore/preprocessor/_bias.py#L88-L89

Reply:
I am aware of that, but thought this might be safer this way. Is it really a good idea to attach the original product as ancestor? Any potential changes (e.g., addition of further ancestors) to this ancestor down the road will also affect the original product, and that just feels wrong. I can already see us having a similar problem in a year when this has been forgotten...

How expensive is this copying anyway? We are not doing this a thousand times, so I don't think it has a big influence on performance.

Reply:
> Is it really a good idea to attach the original product as ancestor? Any potential changes (e.g., addition of further ancestors) to this ancestor down the road will also affect the original product, and that just feels wrong.

Indeed that would be wrong and that's why the current implementation does not do that, `wasderivedfrom` already creates a copy: https://github.com/ESMValGroup/ESMValCore/blob/a4b5bc67d97f76b120403fc51a69536857740695/esmvalcore/_provenance.py#L241.